src/DotVVM.Framework.Tests.Common/ViewModel/ViewModelTypeMetadataSerializerTests.cs

```diff
@@ -0,0 +1,172 @@
+        public void ViewModelTypeMetadata_TypeName(Type type, string expected)
+        {
+            var typeMetadataSerializer = new ViewModelTypeMetadataSerializer();
+            var result = typeMetadataSerializer.GetTypeIdentifier(type);
+            Assert.AreEqual(expected.Replace("'", "\""), result.ToString(Formatting.None));
+        }
+
+        [TestMethod]        
+        public void ViewModelTypeMetadata_TypeMetadata()
+        {
+            var typeMetadataSerializer = new ViewModelTypeMetadataSerializer();
+            var result = typeMetadataSerializer.SerializeTypeMetadata(new[]
+            {
+                mapper.GetMap(typeof(TestViewModel)),
+                mapper.GetMap(typeof(NestedTestViewModel))
+            });
+
+            var expected = @"{
+    'hXgHohhzHL2SzMTI0/aQFP42rV8=': {
+        'ChildFirstRequest': {
+            'type': 'GqKC8CuCoDNJdqzKjKrxePbUZr8=',
+            'post': 'no',
+            'update': 'firstRequest'
+        },
+        'ClientToServer': {
+            'type': 'String',
+            'update': 'no'
+        },
+        'NestedList': {
+            'type': [
+                'GqKC8CuCoDNJdqzKjKrxePbUZr8='
+            ]
+        },
+        'property ONE': {
+            'type': 'Guid'
+        },
+        'property TWO': {
+            'type': [
+                {
+                    'type': 'nullable',
+                    'inner': {
+                        'type': 'enum',
+                        'values': {
+                            'Zero': 0,
+                            'One': 1,
+                            'Two': 2
+                        }
+                    }
+                }
+            ]
+        },
+        'ServerToClient': {
+            'type': 'String',
+            'post': 'no',
+            'validationRules': [
+                {
+                    'ruleName': 'required',
+                    'errorMessage': 'ServerToClient is required!',
+                    'parameters': []
+                }
+            ]
+        }
+    },
+    'GqKC8CuCoDNJdqzKjKrxePbUZr8=': {
+        'Ignored': {
+            'type': 'Boolean',
+            'post': 'no',
+            'update': 'no'
+        },
+        'InPathOnly': {
+            'type': 'Int32',
+            'post': 'pathOnly',
+            'validationRules': [
+                {
+                    'ruleName': 'required',
+                    'errorMessage': 'The InPathOnly field is required.',
+                    'parameters': []
+                },
+                {
+                    'ruleName': 'range',
+                    'errorMessage': 'range error',
+                    'parameters': [
+                        0,
+                        10
+                    ]
+                }
+            ]
+        }
+    }
+}";
+            Assert.AreEqual(JToken.Parse(expected.Replace("'", "\"")).ToString(Formatting.None), result.ToString(Formatting.None));
```

Comment:
You can use https://github.com/exyi/CheckTestOutput instead of these scary literals ;)